LanguageFeatures/Patterns/Exhaustiveness/lifting_identifier_pattern_A01_t01.dart

```diff
@@ -0,0 +1,40 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion The lifted space union for a pattern with matched value type M is
+/// ...
+/// Variable pattern or identifier pattern: The lifted space union of the static
+/// type of the corresponding variable.
+///
+/// @description Check that a lifted space for an identifier pattern is a union
+/// of the static type of the corresponding variable.
```

Comment:
I believe that's not true (the spec should be adjusted here): In a matching context, an identifier pattern is a reference to a constant variable, and the space for that pattern is a pattern with this type _and_ a constant constraint denoting that particular constant (somehow, perhaps just by means of the variable itself).

However, we can't really observe the space anyway.

I think the test is OK, but we should note that the description should be different, and the spec should be adjusted.